glib/src/param_spec.rs

```diff
@@ -501,6 +501,36 @@ macro_rules! define_builder {
                 self
             }
 
+            /// Mark the property as construct
+            pub fn construct(mut self) -> Self {
+                self.flags |= crate::ParamFlags::CONSTRUCT;
+                self
+            }
+
+            /// Mark the property as construct only
+            pub fn construct_only(mut self) -> Self {
+                self.flags |= crate::ParamFlags::CONSTRUCT_ONLY;
+                self
+            }
+
+            /// Mark the property as lax validation
+            pub fn lax_validation(mut self) -> Self {
+                self.flags |= crate::ParamFlags::LAX_VALIDATION;
+                self
+            }
+
+            /// Mark the property as explicit notify
+            pub fn explicit_notify(mut self) -> Self {
+                self.flags |= crate::ParamFlags::EXPLICIT_NOTIFY;
+                self
+            }
+
+            /// Mark the property as deprecated
+            pub fn deprecated(mut self) -> Self {
```

Comment:
Maybe the builders should all implement a trait so that in gstreamer-rs I can implement an extension trait on all the builders that adds builder methods for all the GStreamer pspec flags.

Reply:
Yeah that would be great indeed, I wanted to port the gst-plugins-rs to the builder patterns, but as the flags has to be set manually, I dropped that idea for now.

Reply:
That trait would have all those common builder functions like `flags()`, the ones you just added, etc. so that I can do a `impl<T: ThatTrait> GStreamerPSpecBuilderExt for T { ... }`

Reply:
That works indeed, but Imho that's too much complexity for something that's already doable by OR-ing the ParamFlags.

Reply:
it is not as you would have to use the TBuilder::flags() instead of using one of the setter methods above. We were already thinking of moving all the builder patterns to traits anyway, so it would make sense to have that here as well for the sake of consistency.

Reply:
Also it's nicer to read compared to the flags.

Reply:
I implemented the trait split